### ISSUE TITLE
`AppendSparseBaseMap` breaks sparse (or is incompatible)

### ISSUE THREAD

**Issue by pgrete** [parthenon-hpc-lab/parthenon#638]
In the process of getting #586 into `develop` it turns out, that the changes introduced with #616 are not compatible.

```bash
$ ./example/sparse_advection/sparse_advection-example -i /home/pgrete/src/parthenon/tst/regression/test_suites/sparse_advection/parthinput.sparse_advection

[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[Detaching after fork from child process 147118]
[New Thread 0x7ffff718b640 (LWP 147122)]
[New Thread 0x7ffff67da640 (LWP 147123)]
#Variables in use:
# Package: parthenon::resolved_state
# ---------------------------------------------------
# Variables:
# Name	Metadata flags
# ---------------------------------------------------
# ---------------------------------------------------
# Sparse Variables:
# Name	sparse id	Metadata flags
# ---------------------------------------------------
sparse_0                  Cell,Provides,Real,Independent,Sparse,FillGhost,WithFluxes
sparse_1                  Cell,Provides,Real,Independent,Sparse,FillGhost,WithFluxes
sparse_2                  Cell,Provides,Real,Independent,Sparse,FillGhost,WithFluxes
sparse_3                  Cell,Provides,Real,Independent,Sparse,FillGhost,WithFluxes
# ---------------------------------------------------
# Swarms:
# Swarm	Value	metadata
# ---------------------------------------------------

### Warning in Mesh::Initialize
The number of MeshBlocks increased more than twice during initialization.
More computing power than you expected may be required.
### Warning in Mesh::Initialize
The number of MeshBlocks increased more than twice during initialization.
More computing power than you expected may be required.

Setup complete, executing driver...

cycle=0 time=0.0000000000000000e+00 dt=3.3145630368119419e-03 zone-cycles/wsec_step=0.00e+00 wsec_total=2.66e-02 wsec_step=1.47e+00 zone-cycles/wsec=0.00e+00 wsec_AMR=0.00e+00
---------------------- Current Mesh structure ----------------------
Root grid = 8 x 8 x 1 MeshBlocks
Total number of MeshBlocks = 220
Number of physical refinement levels = 2
Number of logical  refinement levels = 5
  Physical level = 0 (logical level = 3): 48 MeshBlocks, cost = 48
  Physical level = 1 (logical level = 4): 28 MeshBlocks, cost = 28
  Physical level = 2 (logical level = 5): 144 MeshBlocks, cost = 144
--------------------------------------------------------------------
terminate called after throwing an instance of 'std::runtime_error'
  what():  ### PARTHENON ERROR
  Condition:   it != varMap_.end()
  Message:     Couldn't find variable 'sparse'
  File:        /home/pgrete/src/parthenon/src/interface/meshblock_data.hpp
  Line number: 147


Thread 1 "sparse_advectio" received signal SIGABRT, Aborted.
0x00007ffff74e3d22 in raise () from /usr/lib/libc.so.6
(gdb) where
#0  0x00007ffff74e3d22 in raise () from /usr/lib/libc.so.6
#1  0x00007ffff74cd862 in abort () from /usr/lib/libc.so.6
#2  0x00007ffff7748802 in __gnu_cxx::__verbose_terminate_handler () at /build/gcc/src/gcc/libstdc++-v3/libsupc++/vterminate.cc:95
#3  0x00007ffff7754c8a in __cxxabiv1::__terminate (handler=<optimized out>) at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:48
#4  0x00007ffff7754cf7 in std::terminate () at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:58
#5  0x00007ffff7754f8e in __cxxabiv1::__cxa_throw (obj=<optimized out>, tinfo=0x7ffff78b7220 <typeinfo for std::runtime_error>, dest=0x7ffff776b050 <std::runtime_error::~runtime_error()>)
    at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_throw.cc:95
#6  0x00005555555abb34 in parthenon::ErrorChecking::require_throws (condition=0x55555582f1ec "it != varMap_.end()", message=0x555557499490 "Couldn't find variable 'sparse'", 
    filename=0x55555582ca25 "/home/pgrete/src/parthenon/src/interface/meshblock_data.hpp", linenumber=147) at /home/pgrete/src/parthenon/src/utils/error_checking.hpp:138
#7  0x00005555555b320b in parthenon::ErrorChecking::require_throws (condition=0x55555582f1ec "it != varMap_.end()", message="Couldn't find variable 'sparse'", 
    filename=0x55555582ca25 "/home/pgrete/src/parthenon/src/interface/meshblock_data.hpp", linenumber=147) at /home/pgrete/src/parthenon/src/utils/error_checking.hpp:143
#8  0x00005555555d1184 in parthenon::MeshBlockData<double>::GetCellVarPtr (this=0x555555d4b270, label="sparse") at /home/pgrete/src/parthenon/src/interface/meshblock_data.hpp:146
#9  0x0000555555610000 in parthenon::MeshBlockData<double>::Get (this=0x555555d4b270, base_name="sparse", sparse_id=-2147483648)
    at /home/pgrete/src/parthenon/src/interface/meshblock_data.hpp:153
#10 0x000055555565548c in parthenon::Update::SparseDealloc (md=0x5555560a5050) at /home/pgrete/src/parthenon/src/interface/update.cpp:208
#11 0x000055555559d20c in parthenon::TaskList::AddTask<parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*>(parthenon::TaskID const&, parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*&&)::{lambda()#1}::operator()() (this=0x5555575b7ec0) at /home/pgrete/src/parthenon/src/tasks/task_list.hpp:300
#12 0x000055555559d1bd in std::__invoke_impl<parthenon::TaskStatus, parthenon::TaskList::AddTask<parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*>(parthenon::TaskID const&, parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*&&)::{lambda()#1}&>(std::__invoke_other, parthenon::TaskList::AddTask<parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*>(parthenon::TaskID const&, parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*&&)::{lambda()#1}&) (__f=...) at /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/bits/invoke.h:61
#13 0x000055555559d15d in std::__invoke_r<parthenon::TaskStatus, parthenon::TaskList::AddTask<parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*>(parthenon::TaskID const&, parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*&&)::{lambda()#1}&>(parthenon::TaskList::AddTask<parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*>(parthenon::TaskID const&, parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*&&)::{lambda()#1}&) (__fn=...)
    at /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/bits/invoke.h:142
#14 0x000055555559d04d in std::_Function_handler<parthenon::TaskStatus (), parthenon::TaskList::AddTask<parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*>(parthenon::TaskID const&, parthenon::TaskStatus (&)(parthenon::MeshData<double>*), parthenon::MeshData<double>*&&)::{lambda()#1}>::_M_invoke(std::_Any_data const&) (__functor=...)
    at /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/bits/std_function.h:291
#15 0x00005555555fa585 in std::function<parthenon::TaskStatus ()>::operator()() const (this=0x5555575b7ec0)
    at /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/bits/std_function.h:560
#16 0x00005555555f9cad in parthenon::Task::operator() (this=0x5555575b7e80) at /home/pgrete/src/parthenon/src/tasks/task_types.hpp:67
#17 0x00005555555f7af9 in parthenon::TaskList::DoAvailable (this=0x5555561ac890) at /home/pgrete/src/parthenon/src/tasks/task_list.hpp:244
#18 0x00005555555f7875 in parthenon::TaskCollection::Execute (this=0x7fffffffe308) at /home/pgrete/src/parthenon/src/tasks/task_list.hpp:445
#19 0x00005555555f663f in parthenon::DriverUtils::ConstructAndExecuteTaskLists<parthenon::MultiStageDriver, int> (driver=0x7fffffffe508, args=2)
    at /home/pgrete/src/parthenon/src/driver/driver.hpp:115
#20 0x00005555555f624c in parthenon::MultiStageDriver::Step (this=0x7fffffffe508) at /home/pgrete/src/parthenon/src/driver/multistage.cpp:93
#21 0x00005555555f3dfa in parthenon::EvolutionDriver::Execute (this=0x7fffffffe508) at /home/pgrete/src/parthenon/src/driver/driver.cpp:75
#22 0x00005555555be640 in main (argc=3, argv=0x7fffffffe6e8) at /home/pgrete/src/parthenon/example/sparse_advection/main.cpp:49
```

Given that I don't have access to the downstream example in which #616 was tested, I temporarily disabled `AppendSparseBaseMap(vars, pvmap);` in `variable_pack.hpp` assuming that it'll be a straightforward fix on your end as you have the code available, @jdolence  @lroberts36

**Comment by jdolence** [comment]
I'll take a look at this

**Comment by pgrete** [comment]
Closed with #586
